### PR TITLE
Add backend server

### PR DESCRIPTION
**Pull request by bobbiec**
This PR adds a backend server. 

Currently, it serves a fixed list of donuts at `/api/donuts`. 

The menu page now loads the list of donuts from the server, displaying a spinner during the request.

### CODE REVIEW

File: src/components/inventory.tsx [hunk 1]
```diff
@@ -1,157 +1,155 @@
 import * as React from 'react';
-import { Component, ChangeEvent, FormEvent } from 'react';
+import { ChangeEvent, FormEvent } from 'react';
```
Comment (ykotturi) [inline src/components/inventory.tsx:2]: Is this greenkeeper who is making all of these what look like linting edits to this inventory file?

File: src/components/menu.tsx [hunk 2]
```diff
@@ -18,26 +29,53 @@ export interface MenuProps {
...
-    };
+  async componentDidMount() {
+    setTimeout(async () => {
+      const items: MenuItemProps[] = await getDonuts();
+      this.setState((prevState) => ({
+          items: items,
+          cart: this.initCart(items)
+        })
+      );
+      console.log(items);
```
Comment (ykotturi) [inline src/components/menu.tsx:78]: Did you want this console.log() in master?

File: src/router.tsx [hunk 3]
```diff
@@ -15,11 +15,7 @@ export const AppRouter: React.StatelessComponent<{}> = () => {
               {id: "2", name: "Chocolate Glazed", ingredients: "white_flour, chocolate, butter", priceInCents: 150, inStock: true,}]} /> } />
           <Route path="/hello3" render={(props) => <Hello {...props} name="page 3" /> } />
           <Route path="/menu" render={(props) =>
-            <Menu {...props} items={[
-              {id: "1", name: "Original Glazed", priceInCents: 100},
-              {id: "2", name: "Chocolate Glazed", priceInCents: 150, imageUrl: "https://cmu-17-356.github.io/Dronuts/assets/donut_flavors/chocolate_glaze.jpg"},
-              {id: "3", name: "Jelly", priceInCents: 200, imageUrl: "https://cmu-17-356.github.io/Dronuts/assets/donut_flavors/jelly.jpg"},
-            ]} />
+            <Menu {...props} items={[]} />
```
Comment (ykotturi) [inline src/router.tsx:27]: A question for you: before we merge to master do we all need to apply this same logic to each of our views? Specifically, creating donuts in server/index.js VS here in the router?

**Comment by ykotturi** [comment]
Overall: I pulled this branch and tested locally - looks good and I especially like the spinning donut while view loads. Maybe @shumry might also review this PR as I had more questions than constructive comments.

File: server/index.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+app.use(bodyParser.urlencoded({ extended: false }));
+
+app.get('/api/greeting', (req, res) => {
+  const name = req.query.name || 'World';
+  res.setHeader('Content-Type', 'application/json');
+  res.send(JSON.stringify({ greeting: `Hello ${name}!` }));
+});
+
+app.get('/api/donuts', (_, res) => {
+  const donuts = [
```
Comment (shumry) [inline server/index.js:14]: More of a question than a comment, when we want to edit the donut dictionary, will that code be written in this file?

File: src/components/helpers/index.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  let response = await promise;
+  let result = await response.json();
+  return result;
+}
+
+export function getDonuts() {
+  const getDonutUrl = '/api/donuts';
+  return getUrl(getDonutUrl);
+}
```
Comment (shumry) [inline src/components/helpers/index.ts:30]: This modularity is a good idea!

File: src/components/menu.tsx [hunk 6]
```diff
@@ -290,11 +311,4 @@ async function createTransaction(totalPrice: number) {
...
 }
 
-async function getUrl(url: string) {
-  let promise = fetch(url);
-  let response = await promise;
-  let result = await response.json();
-  return result;
-}
-
 export default Menu;
```
Comment (shumry) [inline src/components/menu.tsx:314]: Thanks for cleaning up the whitespace! Will do a better job with it next time.

**Pull request merged**